zonemaster/zonemaster-gui
# Upgrade angular and bootstrap

## Purpose

This PR upgrade to Angular 13 and Bootstrap 5.

## Context

* Angular 12 EOL in November 2022.
* ng-bootstrap ties the angular version to the bootstrap version. While we could have waited for Angular 14 to upgrade as there is one version of overlap to ease the migration I took the opportunity to upgrade it tool.

~~Note: I am marking this PR as draft because of #348~~ Migration of the i18n system has been included in the PR.

## Changes

* [x] Upgrade of angular and bootstrap and some minor dependencies.
* [x] Some dev dependencies have been removed as they are unused.
* [x] Migration to angular/localize (implies some changes on how language change is handle)
* [x] Remove all mention of ngx-translated.
* [x] Update e2e tests with new URL, add small dev server to serve all locales (angular dev server can only serve one local at a time)
* [x] Add documentation for translation
* [x] Update configuration (default language is now irrelevant in app config)
* [x] Update Apache example configuration
* [x] Update instruction to serve GUI from different base URL
* [x] Add instruction to use translation tools (poedit only)

During the migration to angular/localize, I tagged some strings that were not translated so that they could be translated.
Note: once this PR is merged we should create some issues to review the new translation files.

## How to test this PR

The GUI should behave as usual.

Known behaviour changes:
Changing the language reload the page causing:
* the domain form to be closed and reset if the user was editing it at the time of change
* the results modules to be collapsed if they were expanded

The Apache configuration has been updated to work correctly with the new release bundle architecture.

Without using Apache a simple server can be used to use the localized bundle by running `npm run e2e:test-server` (test configuration that uses zonemaster.net api).
